pkg/api/v2/reporter/reporter.go

```diff
@@ -192,34 +193,118 @@ func (e ResourceReports) FilterByKind(kind string) ResourceReports {
 	return reports
 }
 
-// ignores warnings
+// refMapAndSortedKeys returns a map of resource refs to resources and a sorted list of resource refs
+// This is used to iterate over the resources in a consistent order.
+// The reports are keyed by references to the resources, so are not sortable.
+// There is no unique key for a resource, so we use the string representation of the name/namespace as the key,
+// and collect all the resources with the same key together.
+// The previous implementation did not guarantee a consistent order when iterating over the resources,
+// so any order used here will be acceptable for backwards compatibility.
+func (e ResourceReports) refMapAndSortedKeys() (map[string][]resources.InputResource, []string) {
+	// refKeys contains all the unique keys for the resources
+	var refKeys []string
+	// refMaps is a map of resource keys to a slice of resources with that key
+	var refMap = make(map[string][]resources.InputResource)
+
+	// Loop over the resources
+	for res := range e {
+
+		// Get a string representation of the resource ref. This is not guaranteed to be unique.
+		resKey := res.GetMetadata().Ref().String()
+
+		// Add the key to the list of keys if it is not already there
+		if !slices.Contains(refKeys, resKey) {
```

Comment:
nit: Instead of keeping the refKeys and the map, and checking if the key exists in the refKeys before adding it, would it just be simpler to get the refKeys after the map has been populated ?